plugin/src/main/cpp/extensions/openxr_fb_space_warp_extension_wrapper.cpp

```diff
@@ -0,0 +1,289 @@
+
+		space_warp_info[i].depthSubImage.swapchain = (XrSwapchain)get_openxr_api()->openxr_swapchain_get_swapchain(motion_vector_depth_swapchain_info);
+		space_warp_info[i].depthSubImage.imageRect.offset.x = 0;
+		space_warp_info[i].depthSubImage.imageRect.offset.y = 0;
+		space_warp_info[i].depthSubImage.imageRect.extent.width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+		space_warp_info[i].depthSubImage.imageRect.extent.height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+		space_warp_info[i].depthSubImage.imageArrayIndex = i;
+
+		space_warp_info[i].minDepth = 0.0;
+		space_warp_info[i].maxDepth = 1.0;
+
+		space_warp_info[i].farZ = get_openxr_api()->get_render_state_z_near();
+		space_warp_info[i].nearZ = get_openxr_api()->get_render_state_z_far();
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_pre_render() {
+	if (!fb_space_warp_ext || !enabled) {
+		return;
+	}
+
+	get_openxr_api()->openxr_swapchain_acquire(motion_vector_swapchain_info);
+	get_openxr_api()->openxr_swapchain_acquire(motion_vector_depth_swapchain_info);
+
+	RID motion_vector_swapchain_image = get_openxr_api()->openxr_swapchain_get_image(motion_vector_swapchain_info);
+	get_openxr_api()->set_velocity_texture(motion_vector_swapchain_image);
+	RID motion_vector_depth_swapchain_image = get_openxr_api()->openxr_swapchain_get_image(motion_vector_depth_swapchain_info);
+	get_openxr_api()->set_velocity_depth_texture(motion_vector_depth_swapchain_image);
+
+	int target_width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+	int target_height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+	Size2i render_target_size = { target_width, target_height };
+	get_openxr_api()->set_velocity_target_size(render_target_size);
+
+	Transform3D world_transform = XRServer::get_singleton()->get_world_origin();
+	Transform3D delta_transform = render_state.previous_transform.affine_inverse() * world_transform;
+	Quaternion delta_quat = delta_transform.basis.get_quaternion();
+	Vector3 delta_origin = delta_transform.origin;
+
+	Ref<OpenXRInterface> openxr_interface = XRServer::get_singleton()->find_interface("OpenXR");
+	int view_count = openxr_interface->get_view_count();
+	for (int i = 0; i < view_count; i++) {
+		space_warp_info[i].layerFlags = render_state.skip_space_warp_frame ? XR_COMPOSITION_LAYER_SPACE_WARP_INFO_FRAME_SKIP_BIT_FB : 0;
+		space_warp_info[i].appSpaceDeltaPose = { { delta_quat.x, delta_quat.y, delta_quat.z, delta_quat.w }, { delta_origin.x, delta_origin.y, delta_origin.z } };
+		space_warp_info[i].farZ = get_openxr_api()->get_render_state_z_near();
+		space_warp_info[i].nearZ = get_openxr_api()->get_render_state_z_far();
+	}
+
+	render_state.skip_space_warp_frame = false;
+	render_state.previous_transform = world_transform;
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_post_draw_viewport(const RID &p_render_target) {
+	if (!fb_space_warp_ext || !enabled) {
```

Comment:
Here and elsewhere, it looks like we can replace this logic with `!is_enabled()`